src/renderer/pandoc/mod.rs

```diff
@@ -0,0 +1,42 @@
+use renderer::Renderer;
+use book::MDBook;
+use book::bookitem::BookItem;
+
+use std::error::Error;
+use std::process::Command;
+
+pub struct Pandoc;
+
+impl Pandoc {
+  pub fn new() -> Pandoc {
+    Pandoc
+  }
+}
+
+impl Renderer for Pandoc {
+    fn render(&self, book: &MDBook) -> Result<(), Box<Error>> {
+        let mut paths = vec!();
+
+        for item in book.iter() {
+            match *item {
+                BookItem::Chapter(_, ref ch) => {
+                    paths.push(book.get_src().join(&ch.path).into_os_string());
+                },
+                _ => println!("FIXME: don't understand this kind of BookItem")
+            }
+        }
+
+        let output = Command::new("pandoc")
+            .arg("-S")
+            .arg("-osample.epub")
+            .args(&paths)
+            .output();
+
+        match output {
+            Ok(_) => Ok(()),
+            Err(e) => Err(Box::new(e))
+        }
+        // FIXME: why doesn't this work
+        // output.map(|_| ()).map_err(|e| Box::new(e))
```

Comment:
You don't need to be afraid to use the `match` statement, it is not considered "a last resort bazooka" that should be avoided when possible. Generally there are multiple ways to handle a certain pattern, in this case the match works. You could also have written:

```rust
if let Err(e) = output {
    return Err(Box::new(e))
}

Ok(())
```

Maybe there are other solutions, but it is generally a matter of personal preference :)